src/geometry/aabb.rs

```diff
@@ -1,72 +1,170 @@
+    pub fn axes() -> [Unit<Vector3<S>>; 3] {
+        [Vector3::x_axis(), Vector3::y_axis(), Vector3::z_axis()]
+    }
+}
+
+impl From<&proto::AxisAlignedCuboid> for Aabb<f64> {
+    fn from(aac: &proto::AxisAlignedCuboid) -> Self {
+        let aac_min = aac.min.clone().unwrap_or_else(|| {
+            let deprecated_min = aac.deprecated_min.clone().unwrap(); // Version 9
+            proto::Vector3d::from(deprecated_min)
+        });
+        let aac_max = aac.max.clone().unwrap_or_else(|| {
+            let deprecated_max = aac.deprecated_max.clone().unwrap(); // Version 9
+            proto::Vector3d::from(deprecated_max)
+        });
+        Aabb::new(
+            std::convert::From::from(&aac_min),
+            std::convert::From::from(&aac_max),
+        )
+    }
+}
+
+impl From<&Aabb<f64>> for proto::AxisAlignedCuboid {
+    fn from(bbox: &Aabb<f64>) -> Self {
+        let mut aac = proto::AxisAlignedCuboid::new();
+        aac.set_min(proto::Vector3d::from(bbox.min()));
+        aac.set_max(proto::Vector3d::from(bbox.max()));
+        aac
+    }
+}
+
+impl<S> PointCulling<S> for Aabb<S>
 where
-    S: 'static + BaseFloat + Sync + Send + Bounded,
+    S: RealField + num_traits::Bounded,
 {
     fn contains(&self, p: &Point3<S>) -> bool {
-        Contains::contains(self, p)
+        self.contains(p)
     }
 
-    fn intersects_aabb3(&self, aabb: &Aabb3<S>) -> bool {
+    fn intersects_aabb(&self, aabb: &Aabb<S>) -> bool {
         sat::sat(
-            ArrayVec::from([Vector3::unit_x(), Vector3::unit_y(), Vector3::unit_z()]),
-            &self.to_corners(),
-            &aabb.to_corners(),
+            ArrayVec::from([Vector3::x_axis(), Vector3::y_axis(), Vector3::z_axis()]),
+            &self.compute_corners(),
+            &aabb.compute_corners(),
         ) != Relation::Out
     }
 }
 
-impl<S> ConvexPolyhedron<S> for Aabb3<S>
+impl<S> ConvexPolyhedron<S> for Aabb<S>
 where
-    S: BaseFloat,
+    S: RealField,
 {
     fn compute_corners(&self) -> [Point3<S>; 8] {
-        self.to_corners()
+        [
+            Point3::new(self.mins.x, self.mins.y, self.mins.z),
+            Point3::new(self.maxs.x, self.mins.y, self.mins.z),
+            Point3::new(self.mins.x, self.maxs.y, self.mins.z),
+            Point3::new(self.maxs.x, self.maxs.y, self.mins.z),
+            Point3::new(self.mins.x, self.mins.y, self.maxs.z),
+            Point3::new(self.maxs.x, self.mins.y, self.maxs.z),
+            Point3::new(self.mins.x, self.maxs.y, self.maxs.z),
+            Point3::new(self.maxs.x, self.maxs.y, self.maxs.z),
+        ]
     }
 
     fn intersector(&self) -> Intersector<S> {
         let mut unit_axes = ArrayVec::new();
-        unit_axes.push(Vector3::unit_x());
-        unit_axes.push(Vector3::unit_y());
-        unit_axes.push(Vector3::unit_z());
-        let corners = self.compute_corners();
+        unit_axes.push(Vector3::x_axis());
+        unit_axes.push(Vector3::y_axis());
+        unit_axes.push(Vector3::z_axis());
```

Comment:
how about:
```suggestion
unit_axes
            .try_extend_from_slice(&[Vector3::x_axis(), Vector3::y_axis(), Vector3::z_axis()])
            .unwrap();
```

Reply:
Tbh I'd prefer what we have now since it's easier to understand. Let me know if you see a reason why it should use the slice method.